### PR TITLE
Bifrost mpos overhaul

### PR DESCRIPTION
**Pull request by darakeon**
Improvements made during Bifrost development that should stay at this repository, because changes the SDK.

### CODE REVIEW

File: PagarMe.Mpos/Abecs/AbecsStream.cs [hunk 1]
```diff
@@ -97,17 +100,22 @@ protected virtual void Dispose(bool disposing)
...
+                return;
+
             var buffer = new byte[2048];
 
             BaseStream.ReadAsync(buffer, 0, buffer.Length).ContinueWith(t =>
             {
                 NativeStream->DataReceived(NativeStream, buffer, t.Result);
-
                 BeginRead();
+
```
Comment (wilkmaia) [inline PagarMe.Mpos/Abecs/AbecsStream.cs]: Please remove this empty line.

File: PagarMe.Mpos/Devices/DeviceManager.cs [hunk 2]
```diff
@@ -52,14 +52,14 @@ private Task updateTask()
 
         private void update()
         {
-            var serialPorts = SerialPort.GetPortNames();
+            var serialPorts = SerialPort.GetPortNames();
```
Comment (wilkmaia) [inline PagarMe.Mpos/Devices/DeviceManager.cs]: Please remove trailing spaces.

File: PagarMe.Generic/MakeArrayExtension.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace PagarMe.Generic
+{
+    public static class MakeArrayExtension
```
Comment (wilkmaia) [inline PagarMe.Generic/MakeArrayExtension.cs]: Where is this class used?

**Comment by darakeon** [comment]
All fixed. Removed other unused classes

**Comment by darakeon** [comment]
All fixed. Removed other unused classes

**Comment by darakeon** [comment]
@wilkmaia , did you approved this?

**Comment by wilkmaia** [comment]
Now I did :)

**Pull request merged**